src/main/java/tr/com/srdc/cda2fhir/transform/CCDTransformerImpl.java

```diff
@@ -197,23 +214,29 @@ public Bundle transformDocument(ClinicalDocument cda) {
         }
 
         // init the global ccd bundle via a call to resource transformer, which handles cda header data (in fact, all except the sections)
-        Bundle ccdBundle = resTransformer.tClinicalDocument2Composition(ccd);
+        Bundle ccdBundle = resTransformer.tClinicalDocument2Bundle(ccd, includeComposition);
+        
         // the first bundle entry is always the composition
-        Composition ccdComposition = (Composition)ccdBundle.getEntry().get(0).getResource();
+        Composition ccdComposition = includeComposition ? (Composition)ccdBundle.getEntry().get(0).getResource() : null;
+        
         // init the patient id reference if it is not given externally. the patient is always the 2nd bundle entry
-        if (patientRef == null)
-            patientRef = new Reference(ccdBundle.getEntry().get(1).getResource().getId());
-        else // Correct the subject at composition with given patient reference.
+        if (patientRef == null) {
+            patientRef = new Reference(ccdBundle.getEntry().get(1).getResource().getId()); // TO DO: Do not depend on the order
+        } else if (ccdComposition != null) { // Correct the subject at composition with given patient reference.
             ccdComposition.setSubject(patientRef);
-
+        }
+            
         // transform the sections
         for(Section cdaSec: ccd.getSections()) {
             SectionComponent fhirSec = resTransformer.tSection2Section(cdaSec);
             
-            if(fhirSec == null)
+            if(fhirSec == null) {
             	continue;
-            else
+            }
+            
+            if (ccdComposition != null) {
                 ccdComposition.addSection(fhirSec);
```

Comment:
Why are all of the fhir sections added to the composition anyway?

Reply:
It is required in Document type bundles.  See http://hl7.org/fhir/stu3/bundle.html.